src/common/server.ts

```diff
@@ -20,7 +27,50 @@ import { updateStatus } from "./status";
 import { getLSClientTraceLevel, getProjectRoot } from "./utilities";
 import { isVirtualWorkspace } from "./vscodeapi";
 
-export type IInitOptions = { settings: ISettings[]; globalSettings: ISettings };
+export type IInitOptions = {
+  settings: ISettings[];
+  globalSettings: ISettings;
+};
+
+async function createExperimentalServer(
+  settings: ISettings,
+  serverId: string,
+  serverName: string,
+  outputChannel: LogOutputChannel,
+  initializationOptions: IInitOptions,
+): Promise<LanguageClient> {
+  const command = RUFF_BIN_PATH;
+  const cwd = settings.cwd;
+  const args = [RUFF_SERVER_CMD, ...RUFF_SERVER_REQUIRED_ARGS];
+  let env = { ...process.env };
+
```

Comment:
> Or did you decide against supporting that to avoid the situation where the ruff version is too old?

That was the reason, though honestly it shouldn't be too hard to implement a version check here.

Reply:
That makes sense. I'm okay with supporting this functionality in a next version.